manager/src/main/java/io/aklivity/zilla/manager/internal/commands/install/cache/ZpmCache.java

```diff
@@ -15,238 +15,388 @@
+                descriptorRequest.setArtifact(artifact);
+                try
+                {
+                    ArtifactDescriptorResult descriptorResult =
+                        repositorySystem.readArtifactDescriptor(session, descriptorRequest);
+                    final List<Dependency> managedDependencies = descriptorResult.getManagedDependencies();
+                    managedDependencies.forEach(dep ->
+                    {
+                        final Artifact managedArtifact = dep.getArtifact();
+                        imported.put(ZpmDependency.of(managedArtifact.getGroupId(), managedArtifact.getArtifactId(), null),
+                            managedArtifact.getVersion());
+
+                    });
+                }
+                catch (ArtifactDescriptorException e)
+                {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+        CollectRequest collectRequest = new CollectRequest();
+        for (ZpmDependency dep : dependencies)
+        {
+            String version = ofNullable(dep.version).orElse(imported.get(dep));
+            Artifact artifact = new DefaultArtifact(dep.groupId, dep.artifactId, "jar", version);
+            collectRequest.addDependency(new Dependency(artifact, null));
+        }
+        repositories.forEach(collectRequest::addRepository);
+
+        DependencyResult result;
+        try
+        {
+            DependencyRequest dependencyRequest = new DependencyRequest(collectRequest, null);
+            result = repositorySystem.resolveDependencies(session, dependencyRequest);
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException("Failed to resolve dependencies", e);
+        }
+        DependencyNode root = result.getRoot();
+
+        NodeListGenerator nlg = new NodeListGenerator();
+
+        root.accept(new PreorderDependencyNodeConsumerVisitor(nlg));
+        List<DependencyNode> nodesWithDependencies = nlg.getNodesWithDependencies();
+
+        nodesWithDependencies.forEach(node ->
+        {
+            Dependency dep = node.getDependency();
+            if (dep != null)
+            {
+                final Artifact artifact = dep.getArtifact();
+                List<DependencyNode> children = node.getChildren();
+                final ZpmArtifactId id =
+                    new ZpmArtifactId(artifact.getGroupId(), artifact.getArtifactId(), artifact.getVersion());
+                final Set<ZpmArtifactId> depends = new LinkedHashSet<>();
+                children.forEach(c ->
+                {
+                    final Artifact cArtifact = c.getArtifact();
+                    final ZpmArtifactId cid =
+                        new ZpmArtifactId(cArtifact.getGroupId(), cArtifact.getArtifactId(), cArtifact.getVersion());
+                    depends.add(cid);
+                });
+                artifacts.add(new ZpmArtifact(id, artifact.getFile().toPath(), depends));
```

Comment:
Confirming we no longer need the `noneMatch` check and `version` comparison?

Reply:
That's right, we don't need it anymore